dvc/external_repo.py

```diff
@@ -45,117 +36,49 @@ def __init__(self, url, rev, cause):
-        import git
+    if os.name == "nt":
+        # git.exe may hang for a while not permitting to remove temp dir
+        os_retry = retry(5, errors=OSError, timeout=0.1)
+        os_retry(remove)(repo.root_dir)
+    else:
+        remove(repo.root_dir)
 
-        try:
-            git.Repo.clone_from(
-                self.url, tmp_dir, depth=1, no_single_branch=True
-            )
-        except git.exc.GitCommandError as exc:
-            raise InstallError(self.url, tmp_dir, exc)
 
-        if self.rev:
-            try:
-                repo = git.Repo(tmp_dir)
-                repo.git.checkout(self.rev)
-                repo.close()
-            except git.exc.GitCommandError as exc:
-                raise RevError(self.url, self.rev, exc)
-
-        if cache_dir:
-            from dvc.repo import Repo
-
-            repo = Repo(tmp_dir)
-            cache_config = CacheConfig(repo.config)
-            cache_config.set_dir(cache_dir, level=Config.LEVEL_LOCAL)
-            repo.scm.git.close()
-
-    def install(self, cache_dir=None, force=False):
-        if self.installed and not force:
-            logger.info(
-                "Skipping installing '{}'('{}') as it is already "
-                "installed.".format(self.name, self.url)
-            )
-            return
-
-        makedirs(self.repos_dir, exist_ok=True)
-
-        # installing package to a temporary directory until we are sure that
-        # it has been installed correctly.
-        #
-        # Note that we can't use tempfile.TemporaryDirectory is using symlinks
-        # to tmpfs, so we won't be able to use move properly.
-        tmp_dir = os.path.join(self.repos_dir, "." + str(shortuuid.uuid()))
-        try:
-            self._install_to(tmp_dir, cache_dir)
-        except ExternalRepoError:
-            if os.path.exists(tmp_dir):
-                remove(tmp_dir)
-            raise
-
-        if self.installed:
-            self.uninstall()
-
-        shutil.move(tmp_dir, self.path)
-
-    def uninstall(self):
-        if not self.installed:
-            logger.info(
-                "Skipping uninstalling '{}' as it is not installed.".format(
-                    self.name
-                )
-            )
-            return
-
-        # If repo has been initialized then we need to close its git repo
-        if "repo" in self.__dict__:
-            self.repo.scm.git.close()
-
-        if os.name == "nt":
-            # git.exe may hang for a while not permitting to remove temp dir
-            os_retry = retry(5, errors=OSError, timeout=0.1)
-            os_retry(remove)(self.path)
-        else:
-            remove(self.path)
-
-    def update(self):
-        self.repo.scm.fetch(self.rev)
-
-    def dumpd(self):
-        ret = {self.PARAM_URL: self.url}
-
-        if self.rev:
-            ret[self.PARAM_VERSION] = self.rev
-
-        return ret
+@contextmanager
+def external_repo(**kwargs):
+    repo = _install(**kwargs)
```

Comment:
probably install/uninstall names are not appropriate anymore, the same goes for exception names